omod/src/main/java/org/openmrs/module/locationbasedaccess/fragment/controller/manageAccount/AddLocationFragmentController.java

```diff
@@ -42,5 +44,16 @@ else if (person != null) {
                 model.addAttribute("selectedLocationUuid", location.getUuid());
             }
         }
+
+        Object userId = config.getAttribute("userUuid");
+        if (userId != null) {
+            User thisUser = Context.getUserService().getUserByUuid(userId.toString());
+            if (thisUser != null) {
+                String locationProperty = thisUser.getUserProperty(LocationBasedAccessConstants.LOCATION_USER_PROPERTY_NAME);
+                if (StringUtils.isNotBlank(locationProperty)) {
+                    model.addAttribute("selectedLocationUuid", locationProperty);
```

Comment:
For cases where selectedLocationUuid is not set, doesn't the page throw errors?

Reply:
Nope, the `selectedLocationUuid` is already set with a null value at the beginning. So no way to get the NPE here. See [here](https://github.com/openmrs/openmrs-module-locationbasedaccess/blob/7ed3c30e34a05497ef970e70e3e624ebfa693dac/omod/src/main/java/org/openmrs/module/locationbasedaccess/fragment/controller/manageAccount/AddLocationFragmentController.java#L26)